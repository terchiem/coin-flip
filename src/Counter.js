import React, {useState} from 'react';
import CoinResult from './CoinResult';
import Coin from './Coin';

function Counter(props) {
  const [isHeads, setIsHeads] = useState("");
  const [total, setTotal] = useState(0);
  const [heads, setHeads] = useState(0);
  const [tails, setTails] = useState(0);

  function handleClick() {
    const flip = Math.random();
  
    if (flip > 0.5) {
      setHeads(heads + 1);
      setIsHeads(true);
    } else {
      setTails(tails + 1);
      setIsHeads(false)
    }
    setTotal(total + 1);
  };

  const coin = isHeads === "" ? null : <Coin isHeads={isHeads} />;

  return (
    <div>
      <h2>Let's flip a coin!</h2>

      {coin}

      <br></br>
      <button onClick={handleClick}>Flip Coin</button>
      <CoinResult total={total} heads={heads} tails={tails} />
    </div>
  )
};

export default Counter;