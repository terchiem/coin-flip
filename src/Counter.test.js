import React from 'react';
import { render, fireEvent } from "@testing-library/react";
import Counter from './Counter';

it("renders without crashing", function () {
  render(<Counter />);
})

it("matches snapshot", function () {
  const { asFragment } = render(<Counter />);
  expect(asFragment()).toMatchSnapshot();
})

it("starts with no coin image", function () {
  const { queryByTestId } = render(<Counter />);
  const coin = queryByTestId("coin");
  expect(coin).toBe(null);
});


it("updates results when flipping heads and tails", function () {
  jest.spyOn(Math, "random")
    .mockReturnValueOnce(0.25)
    .mockReturnValueOnce(0.75);

  const { getByText } = render(<Counter />);
  const button = getByText("Flip Coin");

  // initial results
  getByText("Out of 0 flips. There have been 0 heads and 0 tails.");

  fireEvent.click(button);

  // results after flipping once
  getByText("Out of 1 flips. There have been 0 heads and 1 tails.");

  fireEvent.click(button);

  // results after flipping twice
  getByText("Out of 2 flips. There have been 1 heads and 1 tails.")
})