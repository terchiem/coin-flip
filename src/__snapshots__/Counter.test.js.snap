// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Let's flip a coin!
    </h2>
    <br />
    <button>
      Flip Coin
    </button>
    <p>
      Out of 0 flips. There have been 0 heads and 0 tails.
    </p>
  </div>
</DocumentFragment>
`;
